### PR TITLE
[WIP] Activity stream

### PR DESCRIPTION
**Pull request by Shine-neko**

### CODE REVIEW

File: src/Tempo/Bundle/AppBundle/EventListener/AccessSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            $route  = $this->router->generate($modelName. '_show', array('slug' => $resource->getFullSlug()));
+        }
+
+        $notification = $this->notificationManager
+            ->create($event->getUserTo(), $message, $data)
+                ->setLink($route);
+
+        $this->notificationManager->save($notification);
+
+        var_dump($event->getName()); exit;
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/EventListener/AccessSubscriber.php]: WTF ?

File: src/Tempo/Bundle/AppBundle/EventListener/ProjectListener.php [hunk 2]
```diff
@@ -30,16 +30,17 @@ public function __construct(RoomManager $roomManager)
     /**
      * @param ProjectEvent $event
      */
-    public function createProject(ProjectEvent $event)
+    public function createProject($event)
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/EventListener/ProjectListener.php]: Base class for Event ?

File: src/Tempo/Bundle/AppBundle/EventListener/AccessSubscriber.php (hunk 1 above)
Comment (Shine-neko) [inline src/Tempo/Bundle/AppBundle/EventListener/AccessSubscriber.php]: forgotten :(

File: src/Tempo/Bundle/AppBundle/Manager/NotificationManager.php [hunk 3]
```diff
@@ -23,12 +23,12 @@ public function findAllByUserAndState($user, $state)
...
-        $room = new Notification();
-        $room->setUser($user);
-        $room->setLink($link);
-        $room->setMessage($message);
+        return (new Notification())
+            ->setUser($user)
+            ->setData($data)
+            ->setMessage($message);
 
         $this->save($room);
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Manager/NotificationManager.php]: $room doesn't exists anymore

File: src/Tempo/Bundle/AppBundle/Twig/Extension/UserExtension.php [hunk 4]
```diff
@@ -11,16 +11,24 @@
...
+    private $translator;
+
+    private $notificationManager;
 
     /**
-     * @param $manager
+     * @param $notificationManager
      */
-    public function __construct($manager)
+    public function __construct(TranslatorInterface $translator, $notificationManager)
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Twig/Extension/UserExtension.php]: Type Hint for $notificationManager

File: src/Tempo/Bundle/AppBundle/EventListener/ActivitySubscriber.php [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    public static function getSubscribedEvents()
+    {
+        return array(
+            TempoAppEvents::ORGANIZATION_ASSIGN_USER => array('accessNotification', 'createActivity'),
+            TempoAppEvents::ORGANIZATION_DELETE_USER => array('accessNotification', 'createActivity'),
+
+            TempoAppEvents::PROJECT_ASSIGN_USER => array('accessNotification', 'createActivity'),
+            TempoAppEvents::PROJECT_DELETE_USER => array('accessNotification', 'createActivity'),
+
+            'tempo.comment.post_create' => array('createActivity')
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/EventListener/ActivitySubscriber.php]: Where is the constant ?

File: src/Tempo/Bundle/AppBundle/Repository/ActivityRepository.php [hunk 6]
```diff
@@ -19,15 +19,15 @@
...
     {
         $query = $this->createQueryBuilder('activity');
         $query
             ->leftJoin('activity.project', 'project')
             ->leftJoin('project.members', 'user')
-            ->where('activity.target = :target')
+            ->where('activity.target IN(:target)')
             ->andWhere('user.user = :user')
-            ->setParameter('target', 'Project')
+            ->setParameter('target', array('Project', 'Comment'))
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Repository/ActivityRepository.php]: Constant here ?

**Pull request merged**